### DISCUSSION TITLE
Linux: Wayland display connection closed by server

### DISCUSSION THREAD

**Discussion by PillTime** [ppy/osu#30647] [Q&A] [answered]
### Type

Crash to desktop

### Bug description

After updating to 2024.1115.1 the game "crashes" after a short while, no matter if playing, loading, or just waiting in the menus. "Crashes" is in quotation marks because the game seems to still be running (sound can be heard and the process is still up), but there's no way to interact with it as the window does not exist.

The title of the issue is derived from one of the last runtime logs.

### Screenshots or videos

[Example of the bug.](https://youtu.be/eWPrTWxhgNk) (video file was too big for github)

### Version

2024.1115.1

### Logs

[logs.zip](https://github.com/user-attachments/files/17775843/logs.zip)

**Comment by bdach** [comment]
wayland mentioned, downgrading to discussion on sight.

Nothing useful in logs, game looks like as if it was hard-killed. Might be broken compositor, may be SDL3 breakage, may just be wayland breakage ~~(which is a tautology)~~.

Maybe another supporting argument for https://github.com/ppy/osu-framework/pull/6426, although a weak one because it's wayland.

**Comment by PillTime** [comment, marked as answer]
Can't replicate in 2024.1115.2
Thanks.

**Comment by PillTime** [comment]
Bug reintroduced in 2025.220.0, but setting `OSU_SDL3=0` as mentioned in https://github.com/ppy/osu/issues/31955#issuecomment-2671053595 allows the game to run properly.

**Reply by Squaduck** [reply]
+1 for the "crash" and `OSU_SDL3=0` as a fix
When running osu! in a console I've seen both of these written to stderr at the time of the error: (Just one of these at a time, not both in one run.)
```
wp_linux_drm_syncobj_surface_v1#52: error 5: explicit sync is used, but no release point is set
```
```
wp_linux_drm_syncobj_surface_v1#52: error 3: explicit sync is used, but no buffer is attached
```
Probably not helpful though.

**Reply by PillTime** [reply]
I did a bit more testing and it seems that using a mouse does not cause this "crash", even if a tablet is plugged in. As soon as the tablet pen is used to move the cursor, it's just a matter of waiting a few seconds until it crashes.
